docs: added documentation on ICS-20 memo field (#4625)
cosmos/ibc-go: docs/apps/transfer/messages.md

- `Sender` is empty.
- `Receiver` is empty.
- `TimeoutHeight` and `TimeoutTimestamp` are both zero.

> - `Memo` is longer than 256 characters.

Reviewer: I don't think this is true?

Author: Oh wow, you're right. But it is true for ICA. I'll fix. https://github.com/cosmos/ibc-go/blob/ec3166d31cd86f138e1c5559f29483c348e0b901/modules/apps/27-interchain-accounts/types/packet.go#L49

Author: Removed this line

This message will send a fungible token to the counterparty chain represented by the counterparty Channel End connected to the Channel End with the identifiers `SourcePort` and `SourceChannel`.

The denomination provided for transfer should correspond to the same denomination represented on this chain. The prefixes will be added as necessary upon by the receiving chain.

### Memo

> The memo field was added to allow applications and users to attach metadata to transfer packets. The field is optional and can be left empty. When it is used to attach metadata for a particular middleware, the memo field should be represented as a json object where different middlewares add different json keys.

Reviewer: ```suggestion
The memo field was added to allow applications and users to attach metadata to transfer packets. The field is optional and may be left empty. When it is used to attach metadata for a particular middleware, the memo field should be represented as a json object where different middlewares use different json keys.
```

Author: done
